ast.go

```diff
@@ -241,20 +241,16 @@ const (
 
 // CompoundSelect represents a compound operation of selects.
 type CompoundSelect struct {
-	Left    *Select
-	Type    string
-	Right   *Select
-	Limit   *Limit
-	OrderBy OrderBy
+	Left  *Select
+	Type  string
+	Right ReadStatement
```

Comment:
The Right has of the `CompoundSelect` is now a `ReadStatement`, which means it can be a `Select` or another `CompoundSelect`